### PR TITLE
Display question type in assement#show

### LINKED ISSUE #2597: Show question type of each question on the assessment show page

### PR DESCRIPTION
**Pull request by shlok007**
Closes #2597.
updated screenshot:
screenshot from 2018-03-04 01 16 35

### CODE REVIEW

File: app/models/course/assessment/question/text_response.rb [hunk 1]
```diff
@@ -80,6 +80,15 @@ def build_at_least_one_group_one_point
     groups.first.points.build if groups.first.points.empty?
   end
 
+  # returns the type of question as text response or file upload
+  def type
```
Comment (weiqingtoh) [inline app/models/course/assessment/question/text_response.rb]: Actually this method has already been implemented here, see `#question_type`.

**Comment by weiqingtoh** [comment]
Let's have the question type between the Question title and description instead of placing it at the side?

File: app/models/course/assessment/question/text_response.rb (hunk 1 above)
Comment (shlok007) [reply app/models/course/assessment/question/text_response.rb]: I'll use that. Thank you.

File: app/models/course/assessment/question/multiple_response.rb [hunk 2]
```diff
@@ -45,6 +45,15 @@ def initialize_duplicate(duplicator, other)
     self.options = duplicator.duplicate(other.options)
   end
 
+  # returns the type of question as multiple response or multiple choice
+  def type
+    if options.where(correct: true).count > 1
```
Comment (kxmbrian) [inline app/models/course/assessment/question/multiple_response.rb]: Should do the check using `#multiple_choice?` here.

File: app/views/course/question_assessments/_question_assessment.html.slim [hunk 3]
```diff
@@ -17,4 +17,6 @@
             = edit_button(edit_path)
             = delete_button(delete_path)
         h4 = format_inline_text(question_assessment.display_title)
-        = format_html(question.description)
+        span.label.label-primary
+          = format_inline_text(question_assessment.question.actable.type)
```
Comment (weiqingtoh) [inline app/views/course/question_assessments/_question_assessment.html.slim]: Couple of points: 
 1. Shall we just standardise the method name to `#question_type` for all types?
 2. We can make this shorter to `question_asessment.question.question_type`, by adding this  `delegate :question_type, to: :actable`

After this it looks good 👍

File: app/models/course/assessment/question/multiple_response.rb (hunk 2 above)
Comment (shlok007) [reply app/models/course/assessment/question/multiple_response.rb]: I tried using multiple choice instead but the tests kept failing as `#multiple_choice?` is returning `true` even when more than one option is correct.

File: spec/models/course/assessment/question/multiple_response_spec.rb [hunk 4]
```diff
@@ -48,5 +48,26 @@
...
+          create(:course_assessment_question_multiple_response_option, :correct, question: subject)
+        end
+
+        it 'returns multiple response' do
+          expect(subject.type).to eq I18n.t('course.assessment.question.multiple_responses.type.multiple_response')
+        end
+      end
+
+      context 'when question has one correct option' do
+        subject { build(:course_assessment_question_multiple_response) }
```
Comment (weiqingtoh) [inline spec/models/course/assessment/question/multiple_response_spec.rb]: To solve https://github.com/Coursemology/coursemology2/pull/2858#discussion_r172407739, you need to add in the factory trait here, so `build(:course_assessment_question_multiple_response, :multiple_choice)`. You can see more from [`spec/factories/course_assessment_question_multiple_responses.rb`](https://github.com/Coursemology/coursemology2/blob/master/spec/factories/course_assessment_question_multiple_responses.rb#L33-L35).

**Comment by weiqingtoh** [comment]
👍 @shlok007 pls fix the hound errors and LGTM!

File: spec/models/course/assessment/question/multiple_response_spec.rb [hunk 5]
```diff
@@ -48,5 +48,26 @@
...
+        it 'returns multiple response' do
+          expect(subject.question_type).to eq I18n.t('course.assessment.question.multiple_responses.question_type.multiple_response')
+        end
+      end
+
+      context 'when question has only one correct option' do
+        subject { build(:course_assessment_question_multiple_response, :multiple_choice) }
+
+        it 'returns multiple choice' do
+          expect(subject.question_type).to eq I18n.t('course.assessment.question.multiple_responses.question_type.multiple_choice')
```
[the remark this answers is not in the document]
Comment (shlok007) [reply spec/models/course/assessment/question/multiple_response_spec.rb]: Fixed.

File: spec/models/course/assessment/question/multiple_response_spec.rb [hunk 6]
```diff
@@ -48,5 +48,26 @@
...
+
+    describe '#question_type' do
+      context 'when question can have more than one correct option' do
+        subject { build(:course_assessment_question_multiple_response) }
+        let!(:another_correct_option) do
+          create(:course_assessment_question_multiple_response_option, :correct, question: subject)
+        end
+
+        it 'returns multiple response' do
+          expect(subject.question_type).to eq I18n.t('course.assessment.question.multiple_responses.question_type.multiple_response')
```
[the remark this answers is not in the document]
Comment (shlok007) [reply spec/models/course/assessment/question/multiple_response_spec.rb]: Fixed.

**Comment by shlok007** [comment]
Fixed! :slightly_smiling_face:  Shouldn't I squash my commits into one before we proceed to merge?

**Comment by weiqingtoh** [comment]
:+1:

**Pull request merged**